x-pack/plugins/canvas/public/lib/filter.ts

```diff
@@ -53,3 +54,16 @@ export const groupFiltersBy = (filters: FilterType[], groupByField: FilterField)
     filters: groupedFilters[key],
   }));
 };
+
+export const getFiltersByGroups = (filters: string[], groups: string[]) =>
+  filters.filter((filter: string) => {
+    const ast = fromExpression(filter);
+    const expGroups: string[] = get(ast, 'chain[0].arguments.filterGroup', []);
+    return expGroups.length > 0 && expGroups.every((expGroup) => groups.includes(expGroup));
+  });
+
+export const extractGroupsFromElementsFilters = (expr: string) => {
+  const ast = fromExpression(expr);
+  const filtersFn = ast.chain.filter((expression) => expression.function === 'filters')[0];
+  return filtersFn?.arguments.group?.map((g) => g.toString()) ?? [];
```

Comment:
Think there is a bug here if the filter expression is using `ungrouped`.  

Example: make 2 dropdown filters, one with a group, one without.  

Then an elemeent using `filters ungrouped="true"` It will show both filters when it should only be showing the ungrouped filter.

Reply:
Done.